wandb/sdk/launch/agent/agent.py

```diff
@@ -121,25 +127,77 @@ def finish_job_id(self, job_id: Union[str, int]) -> None:
     @staticmethod
     def _parse_runner_config(config: Dict[str, Any]) -> Dict[str, Any]:
         """
-            Parses a RunnersConfig block from the agent config. Returns a dictionary
-            of support runners in the form of:
-            {
-                kubernetes: {
-                    namespace: "wandb"
-                },
-                sagemaker: {
-
-                },
-                local-process: {
-                    labels: [ "gpu-pool" ]
-                },
-                local-container: {
-                    labels: [ "gpu-pool"]
-                }
+        Parses a RunnersConfig block from the agent config. Returns a dictionary
+        of support runners in the form of:
+        {
+            kubernetes: {
+                namespace: "wandb"
+            },
+            sagemaker: {
+
+            },
+            local-process: {
+                labels: [ "gpu-pool" ]
+            },
+            local-container: {
+                labels: [ "gpu-pool"]
             }
+        }
         """
         pass
 
+    def _get_valid_queues(self, entity: str) -> list[RunQueue]:
+        """
+        Given an entity, return a list of run queues associated with that entity.
+        """
+        QUERY = gql(
+            """
+            query GetRunQueuesByEntity($entityName: String!) {
+                runQueues(entityName: $entityName) {
+                    resource_config
+                    queue_id
+                    name
```

Comment:
Just a heads up, we'll need the associated project here as well to pop from run queue